### ISSUE TITLE
Investigate why `AbortNode` message doesn't stop the node

### ISSUE THREAD

**Issue by SanjoDeundiak** [build-trust/ockam#6615]
### Observed behavior

In [this discussion](https://github.com/build-trust/ockam/pull/6445#issuecomment-1760647825) we noticed that sending an `AbortNode` message to the router doesn't cause the node to stop. We should investigate why this happens

**Comment by adiSuper94** [comment]
Just some initial analysis.
```rust
pub enum NodeMessage{
  /// Stop the node (and all workers)
  StopNode(ShutdownType, SmallSender<NodeReplyResult>),
  /// Immediately stop the node runtime
  AbortNode, ...
}
```

```rust
pub enum ShutdownType{
  GraceFul(u8),
  Immediate
}
```
So I greped the code base to see where `AbortNode` is used.
The only place it it used is, in the code which handles graceful shutdown, [here](https://github.com/build-trust/ockam/blob/78180aa2e6156f1d343857f894e87bbc367ac606/implementations/rust/ockam/ockam_node/src/router/shutdown.rs#L92).

The `AbortNode` message aborts the node only if the `RouterState` is `Stopping` [here](https://github.com/build-trust/ockam/blob/78180aa2e6156f1d343857f894e87bbc367ac606/implementations/rust/ockam/ockam_node/src/router/state.rs#L26).

**Comment by SanjoDeundiak** [comment]
@adiSuper94 cool, but does it work as intended in that situation?

**Comment by adiSuper94** [comment]
tl;dr Yes.
So, a little more context. The reason it did not work in #6445 is when we sent the `AbortNode` message was the router state was  still `Running`.  I the case I refer to here, the state is first set to `Stopping`, and then the `AbortNode` message is sent. So it works.

**Comment by SanjoDeundiak** [comment]
@adiSuper94 nice, would you mind submitting a PR with the corresponding comment in the code explaining that?

**Comment by adiSuper94** [comment]
@SanjoDeundiak sure !
